lucene/core/src/java/org/apache/lucene/codecs/lucene90/Lucene90KnnGraphWriter.java

```diff
@@ -216,8 +216,11 @@ private void mergeKnnGraph(FieldInfo mergeFieldInfo, final MergeState mergeState
       int docid;
       while ((docid = sub.nextDoc()) != NO_MORE_DOCS) {
         int mappedDocId = docMap.get(docid);
+        /// deleted document (not alive)
+        if (mappedDocId < 0) {
```

Comment:
You're right, this is needed for document deletion. A small correction: the condition should be `mappedDocId == -1`. (According to the Javadoc of the DocMap#get() method.)

Reply:
Thanks @mocobeta , I have corrected the condition for deleted docIds.